ArchiSteamFarm/Steam/Integration/SteamChatMessage.cs

```diff
@@ -0,0 +1,158 @@
+//     _                _      _  ____   _                           _____
+//    / \    _ __  ___ | |__  (_)/ ___| | |_  ___   __ _  _ __ ___  |  ___|__ _  _ __  _ __ ___
+//   / _ \  | '__|/ __|| '_ \ | |\___ \ | __|/ _ \ / _` || '_ ` _ \ | |_  / _` || '__|| '_ ` _ \
+//  / ___ \ | |  | (__ | | | || | ___) || |_|  __/| (_| || | | | | ||  _|| (_| || |   | | | | | |
+// /_/   \_\|_|   \___||_| |_||_||____/  \__|\___| \__,_||_| |_| |_||_|   \__,_||_|   |_| |_| |_|
+// |
+// Copyright 2015-2021 Łukasz "JustArchi" Domeradzki
+// Contact: JustArchi@JustArchi.net
+// |
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// |
+// http://www.apache.org/licenses/LICENSE-2.0
+// |
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#if NETFRAMEWORK
+using ArchiSteamFarm.Compatibility;
+#endif
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace ArchiSteamFarm.Steam.Integration {
+	internal static class SteamChatMessage {
+		internal const char ContinuationCharacter = '…'; // A character used for indicating that the next newline part is a continuation of the previous line
+		internal const ushort MaxMessageBytes = 6449; // This is a limitation enforced by Steam
+		internal const ushort MaxMessagePrefixBytes = MaxMessageBytes - ReservedContinuationMessageBytes - ReservedEscapeMessageBytes; // Simplified calculation, nobody should be using prefixes even close to that anyway
+		internal const byte NewlineWeight = 61; // This defines how much weight a newline character is adding to the output, limitation enforced by Steam
+		internal const byte ReservedContinuationMessageBytes = 6; // 2x optional … (3 bytes each)
+
+		private const byte ReservedEscapeMessageBytes = 5; // 2 characters total, escape one '\' of 1 byte and real one of up to 4 bytes
+
+		internal static async IAsyncEnumerable<string> GetMessageParts(string message, string? steamMessagePrefix = null) {
```

Comment:
We didn't care previously, so this is not a regression. I classify it as a valid enhancement idea with "PR-ok" label, but I'm not going to tackle that one in this PR, as it didn't work previously and this PR is not meant to make it work.

This is not so hard to implement though, we could calculate number of `:` occurances (as it's special character when used in actual emoticon, not a classic colon, if you use that special character in normal message steam converts it to normal colon, I checked). If the count is odd, we could simply take chars less than `LastIndexOf` it, and move remaining part to the next loop, as we do currently with special encodings.

HOWEVER, this would require bumping current 5 bytes of reserved chars to size of that special semicolon x2 + maximum number of chars steam emoticon can have, that's a lot of work for seemingly random feature, as I don't expect from vanilla ASF to suddenly start sending Steam emoticons.